usmqe_tests/conftest.py

```diff
@@ -348,6 +367,98 @@ def fill_memory():
     return measure_operation(fill_memory)
 
 
+@pytest.fixture(scope="session")
+def volume_mount_points():
+    """
+    Returns dictionary where keys are volume names in gluster and values
+    are directory paths to volume mount points.
+    """
+    SSH = usmssh.get_ssh()
+    host = CONF.inventory.get_groups_dict()["usm_client"][0]
+    gluster_volume = GlusterVolume()
+    volumes = gluster_volume.list()
+    mount_points = {}
+
+    for volume in volumes:
+        mount_point_cmd = "mount | awk '/{}/ {{print $3}}'".format(volume)
+        retcode, mount_point, stderr = SSH[host].run(mount_point_cmd)
+        if retcode != 0:
+            raise OSError(stderr.decode("utf-8"))
+        mount_points[volume] = mount_point.decode("utf-8")
+    return mount_points
+
+
+@pytest.fixture(params=[76, 91], scope="session")
+def workload_capacity_utilization(request, volume_mount_points):
+    """
+    Returns:
+        dict: contains information about `start` and `stop` time of dd
+            command and `result` as number presenting percentage of disk
+            utilization.
+    """
+    volume_name = list(volume_mount_points.keys())[0]
+    mount_point = volume_mount_points[volume_name].strip()
+    SSH = usmssh.get_ssh()
+    host = CONF.inventory.get_groups_dict()["usm_client"][0]
+
+    def fill_volume():
+        """
+        Use `dd` command to utilize mounted volume.
+        """
+        disk_space_cmd = "df {0} | awk '/{1}/ " \
+            "{{print $3 \" \" $4}}'".format(
+                mount_point,
+                mount_point.split("/")[-1])
+        retcode, disk_space, stderr = SSH[host].run(disk_space_cmd)
+        if retcode != 0:
+            raise OSError(stderr.decode("utf-8"))
+
+        # disk values in M
+        disk_used, disk_available = [
+            int(size.rstrip("\n")) / 1024 for size in disk_space.decode(
+                "utf-8").split(" ")]
+
+        # block size = 10M
+        block_size = 10
+        # compute disk space that is going to be used and number of files
+        # to create with regard to already utilized space
+        file_count = int((
+            (int(disk_available) / 100 * request.param) - int(disk_used)
+                ) / block_size)
+
+        stress_cmd = "for x in {{1..{}}}; do dd if=/dev/zero" \
+            " of={}/test_file$x count=1 bs={}M; done".format(
+                file_count,
+                mount_point[:-1] if mount_point.endswith("/") else mount_point,
+                block_size)
```

Comment:
So basically you are creating 10 MB files. That is a lot of files if the capacity is measured in hundreds of GB ...

I would consider tweaking the file size as well, via the `count` option of `dd` based on total capacity of the volume. Creating lot of files to fill the volume is necessary to be able to fill all bricks, but unnecessary granularity could make teardown here too time consuming. Or we could just  try to create 100 MB or 1GB files instead for now and consider the detection later if needed.

Reply:
Note that block size is minimal piece of data dd works with, so it also means that it will allocate that much memory, see:

```
$ /usr/bin/time --verbose dd if=/dev/zero of=zero.100MB bs=100M count=1 2>&1 | grep 'resident set size'
	Maximum resident set size (kbytes): 104516
	Average resident set size (kbytes): 0
$ python3 -c 'print(104532/2**10)'
102.08203125
```

So it would be better to use `bs=10M count=10`, but for 100 MB it would still be ok. But one could be surprised if this is increased again later ...